src/pixelator/plot/__init__.py

```diff
@@ -44,6 +63,652 @@ def _calculate_densities(coordinates, distance_cutoff, unit_sphere_surface):
+        ]
+    ), "Invalid data types"
+
+    if group_by is not None:
+        if group_by not in data.columns:
+            raise ValueError(f"'{group_by}' is missing")
+        assert data[group_by].dtype in [
+            "object",
+            "category",
+        ], "'group_by' must be a character or factor"
+
+    # Define palette
+    palette = {"high": ORANGERED2, "low": SKYBLUE3, "normal": LIGHTGREY}
+
+    # create plot
+    grid = (
+        sns.FacetGrid(
+            data,
+            col=group_by,
+            hue="tau_type",
+            palette=palette,
+            height=4,
+        )
+        if group_by is not None
+        else sns.FacetGrid(data, hue="tau_type", palette=palette, height=4)
+    )
+    grid.map(sns.scatterplot, "tau", "umi_per_upia")
+    grid.set(yscale="log")
+    grid.set_axis_labels("Marker specificity (Tau)", "Pixel content (UMI/UPIA)")
+    grid.add_legend(title="Tau Type")
+
+    return plt.gcf(), plt.gca()
+
+
+def edge_rank_plot(
+    data: pd.DataFrame, group_by: Optional[str] = None
+) -> Tuple[plt.Figure, plt.Axes]:
+    """Plot the number of edges per component against its edge rank.
+
+    :param data: a pandas DataFrame with a column 'edges' containing edge counts for MPX
+    momponents.
+    :param group_by: a column in the DataFrame to group the plot by.
+
+    :return: a plot showing the number of edges per component against its edge rank used
+    for quality control.
+    :rtype: Tuple[plt.Figure, plt.Axes]
+    :raises: AssertionError if the required column(s) are not present in the DataFrame
+    :raises: ValueError if the data types are invalid
+    """
+    assert "edges" in data.columns, "column 'edges' is missing from DataFrame"
+    assert (
+        isinstance(data["edges"], pd.Series) and data["edges"].dtype == int
+    ), "'edges' must be a vector of integer values"
+
+    if group_by is not None:
+        assert (
+            group_by in data.columns
+        ), f"group variable '{group_by}' not found in DataFrame"
+
+        if data[group_by].dtype not in ["object", "category"]:
+            raise ValueError(
+                f"Invalid class '{data[group_by].dtype}' for column '{group_by}'. "
+                f"Expected a string or categorical value"
+            )
+        else:
+            edge_rank_df = data[[group_by, "edges"]].copy()
+            edge_rank_df["rank"] = edge_rank_df.groupby([group_by])["edges"].rank(
+                ascending=False, method="first"
+            )
+    else:
+        edge_rank_df = data[["edges"]].copy()
+        edge_rank_df["rank"] = edge_rank_df["edges"].rank(
+            ascending=False, method="first"
+        )
+
+    (
+        sns.relplot(data=edge_rank_df, x="rank", y="edges", hue=group_by, aspect=1.6)
+        .set(xscale="log", yscale="log")
+        .set_xlabels("Component rank (by number of edges)")
+        .set_ylabels("Number of edges")
+    )
+
+    return plt.gcf(), plt.gca()
+
+
+def cell_count_plot(
```

Comment:
I think maybe we should look into simplifying this function by using `catplot` instead: https://seaborn.pydata.org/generated/seaborn.catplot.html#seaborn.catplot

Reply:
The complexity seems to come from different input permutations that can be used with the function. I looked a bit into catplot and it doesn't seem to be able to simplify our implementation much.